engine.py

```diff
@@ -410,12 +410,75 @@ def post_app_init_nuke(self, menu_name="Shotgun"):
                 # (for example if you do file->open or file->new)
                 tank.util.append_path_to_env_var("NUKE_PATH", app_gizmo_folder)
 
+        # Nuke Studio 9 really doesn't like us starting things at startup, do don't.
+        if not (nuke.env.get("NukeVersionMajor") == 9 and nuke.env.get("studio")):
+            self._run_commands_at_startup()
+
         try:
             self.log_user_attribute_metric("Nuke version", nuke.env.get("NukeVersionString"))
         except:
             # ignore all errors. ex: using a core that doesn't support metrics
             pass
 
+    def _run_commands_at_startup(self):
```

Comment:
All this code until the FIXME is a big copy paste from Maya. Or what it Houdini? (which copied it from Maya in any case). We should create a ticket about refactoring this bit.